service/parserprovider/default_test.go

```diff
@@ -17,123 +17,123 @@ package parserprovider
-		err = cfg.Validate()
-		require.NoError(t, err)
-
-		batch, hasBatch := cfg.Processors[config.NewID("batch")].(*batchprocessor.Config)
-		require.True(t, hasBatch)
-		assert.Equal(t, time.Second*2, batch.Timeout)
-		otlp, hasOTLP := cfg.Receivers[config.NewID("otlp")].(*otlpreceiver.Config)
-		require.True(t, hasOTLP)
-		assert.Equal(t, "localhost:12345", otlp.GRPC.NetAddr.Endpoint)
-	})
+	expectedMap, err := config.NewMapFromBuffer(strings.NewReader(`
+processors:
+  batch:
+exporters:
+  otlp:
+    endpoint: "localhost:4317"`))
+	require.NoError(t, err)
+	assert.Equal(t, expectedMap, cm)
+
+	assert.NoError(t, mp.Close(context.Background()))
 }
 
-func TestDefault_ComponentDoesNotExist(t *testing.T) {
-	factories, err := defaultcomponents.Components()
+func TestDefaultMapProvider_AddNewConfig(t *testing.T) {
+	flags := new(flag.FlagSet)
+	Flags(flags)
+	err := flags.Parse([]string{
+		"--config=testdata/default-config.yaml",
+		"--set=processors.batch.timeout=2s",
+	})
 	require.NoError(t, err)
+	mp := NewDefaultMapProvider()
+	require.NotNil(t, mp)
+	var cm *config.Map
+	cm, err = mp.Get(context.Background())
+	require.NoError(t, err)
+	require.NotNil(t, cm)
+
+	expectedMap, err := config.NewMapFromBuffer(strings.NewReader(`
+processors:
+  batch:
+    timeout: 2s
+exporters:
+  otlp:
+    endpoint: "localhost:4317"`))
+	require.NoError(t, err)
+	assert.Equal(t, expectedMap, cm)
+
+	assert.NoError(t, mp.Close(context.Background()))
+}
 
+func TestDefaultMapProvider_OverwriteConfig(t *testing.T) {
 	flags := new(flag.FlagSet)
 	Flags(flags)
-	err = flags.Parse([]string{
-		"--config=testdata/otelcol-config.yaml",
+	err := flags.Parse([]string{
+		"--config=testdata/default-config.yaml",
 		"--set=processors.batch.timeout=2s",
-		"--set=receivers.otlp.protocols.grpc.endpoint=localhost:12345",
+		"--set=exporters.otlp.endpoint=localhost:1234",
 	})
 	require.NoError(t, err)
+	mp := NewDefaultMapProvider()
+	require.NotNil(t, mp)
+	var cm *config.Map
+	cm, err = mp.Get(context.Background())
+	require.NoError(t, err)
+	require.NotNil(t, cm)
 
-	pl := NewDefaultMapProvider()
-	require.NotNil(t, pl)
-	cp, err := pl.Get(context.Background())
+	expectedMap, err := config.NewMapFromBuffer(strings.NewReader(`
+processors:
+  batch:
+    timeout: 2s
+exporters:
+  otlp:
+    endpoint: "localhost:1234"`))
 	require.NoError(t, err)
-	require.NotNil(t, cp)
-	cfg, err := configunmarshaler.NewDefault().Unmarshal(cp, factories)
+	assert.Equal(t, expectedMap, cm)
+
+	assert.NoError(t, mp.Close(context.Background()))
+}
+
+func TestDefaultMapProvider_InvalidFile(t *testing.T) {
+	flags := new(flag.FlagSet)
+	Flags(flags)
+	err := flags.Parse([]string{
+		"--config=testdata/otelcol-config.yaml",
+	})
+	require.NoError(t, err)
+	mp := NewDefaultMapProvider()
+	require.NotNil(t, mp)
+	_, err = mp.Get(context.Background())
+	require.Error(t, err)
+
+	assert.NoError(t, mp.Close(context.Background()))
+}
+
+func TestDefaultMapProvider_EmptyFile(t *testing.T) {
```

Comment:
Nit: This is really testing a file not set rather than an empty file based on what I'm reading in the test

Reply:
Clarified that is EmptyFileName